catalogue/webapp/components/Calendar/calendar-utils.ts

```diff
@@ -0,0 +1,78 @@
+import moment, { Moment } from 'moment';
+
+export function groupIntoSize(array: any[], size): any[][] {
+  const result = [] as any[];
+  for (let i = 0; i < array.length; i += size) {
+    const group = array.slice(i, i + size);
+    result.push(group);
+  }
+  return result;
+}
+
+export function daysFromStartOfWeek(
+  firstDayOfWeek: number,
+  firstDayOfMonth: number
+): number {
+  if (firstDayOfMonth < firstDayOfWeek) {
+    return 7 - firstDayOfWeek + firstDayOfMonth;
+  } else {
+    return 7 - (7 - firstDayOfMonth) - firstDayOfWeek;
+  }
+}
+
+export function daysUntilEndOfWeek(
+  firstDayOfWeek: number,
+  lastDayOfMonth: number
+): number {
+  const lastDayOfWeek = firstDayOfWeek === 0 ? 6 : firstDayOfWeek - 1;
+  if (lastDayOfMonth > lastDayOfWeek) {
+    return 7 - (lastDayOfMonth - lastDayOfWeek);
+  } else {
+    return lastDayOfWeek - lastDayOfMonth;
+  }
+}
+
+export function getCalendarRows(date: Moment): (Moment | null)[][] {
```

Comment:
Can we avoid using `null`s if possible? There was a time when we aimed to remove `null`s and only used `undefined`s, I feel like if we have a situation where we _need_ both we're probably doing something wrong

Reply:
This is actually out of date, forgot to update the type. I can just remove the null